tests/helium_recovery_plc.py

```diff
@@ -0,0 +1,416 @@
+
+    @parameterized.expand(parameterized_list(COLDBOX_STATUS_VALUES))
+    @skip_if_recsim("lewis backdoor not available in recsim")
+    def test_WHEN_coldbox_status_set_backdoor_THEN_ioc_read_correctly(self, _, test_value):
+        index_test_value = COLDBOX_STATUS_VALUES.index(test_value) + 1
+        self._lewis.backdoor_run_function_on_device("set_memory", ("COLDBOX:STATUS", index_test_value))
+        self.ca.assert_that_pv_after_processing_is("COLDBOX:STATUS", test_value)
+
+    @parameterized.expand(parameterized_list(COLDBOX_STATUS_VALUES))
+    @skip_if_devsim("sim pvs not available in devsim")
+    def test_WHEN_coldbox_status_set_sim_pv_THEN_ioc_read_correctly(self, _, test_value):
+        index_test_value = COLDBOX_STATUS_VALUES.index(test_value)
+        self.ca.set_pv_value("SIM:COLDBOX:STATUS", index_test_value)
+        self.ca.assert_that_pv_after_processing_is("COLDBOX:STATUS", test_value)
+
+    @parameterized.expand(parameterized_list(itertools.product(VALVE_STATUS_PVS, VALVE_STATUS_VALUES)))
+    @skip_if_recsim("lewis backddor not available in recsim")
+    def test_WHEN_valve_status_set_backdoor_THEN_ioc_read_correctly(self, _, pv_name, test_value):
+        index_test_value = VALVE_STATUS_VALUES.index(test_value) + 1
+        self._lewis.backdoor_run_function_on_device("set_memory", (pv_name, index_test_value))
+        self.ca.assert_that_pv_after_processing_is(pv_name, test_value)
+
+    @parameterized.expand(parameterized_list(itertools.product(VALVE_STATUS_PVS, VALVE_STATUS_VALUES)))
+    @skip_if_devsim("sim pvs not available in devsim")
+    def test_WHEN_valve_status_set_sim_pv_THEN_ioc_read_correctly(self, _, pv_name, test_value):
+        index_test_value = VALVE_STATUS_VALUES.index(test_value)
+        self.ca.set_pv_value("SIM:{}".format(pv_name), index_test_value)
+        self.ca.assert_that_pv_after_processing_is(pv_name, test_value)
+
+    # Liquefier alarms are tested separately because in the memory map they are unsigned integers. The C driver does
+    # not support unsigned 16 bit integers directly, but the value is put into a longin record, which should display
+    # the unsigned 16 bit integer correctly. There are two mbbiDirect records that read from the two memory locations
+    # that store alarms in the form of 16 bit integers. These tests then check that the bi records that read from the
+    # mbbiDirect recors work properly.
+
+    @parameterized.expand(parameterized_list(LIQUEFIER_ALARMS))
+    @skip_if_recsim("lewis backdoor not available in recsim")
+    def test_WHEN_liquefier_alarm_set_backdoor_THEN_ioc_read_correctly(self, _, pv_name):
+        alarm_index = LIQUEFIER_ALARMS.index(pv_name)
+        mbbi_direct_pv = HeliumRecoveryPLCTests._get_liquefier_hardware_pv(alarm_index)
+        test_value = HeliumRecoveryPLCTests._get_alarm_test_value(mbbi_direct_pv, alarm_index)
+
+        self.ca.process_pv("{}:_RAW".format(mbbi_direct_pv))
```

Comment:
This string and others are formatted 3 times separately, move out into a variable

Reply:
Done.